### PR TITLE
Network: Consolidate code for generating endpoints

### PR DESCRIPTION
**Pull request by wimrijnders**
This is a preliminary refactoring of the endpoint definitions, in preparation for fixing #3412, #1993 and #3118.

All code related to the drawing of arrow types for edges has been moved to new module `EndPoints` and refactored.

The intention is to make it as simple as possible to create new endpoints. My personal hope is that it becomes so simple, that users take it upon themselves to add new endpoints.

There is room for further simplification; any thoughts on this are welcome.

### CODE REVIEW

**Comment by mbroad** [comment]
Looks good

File: lib/network/modules/components/edges/util/EndPoints.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+   * The following is done:
+   * - rotate by the specified angle
+   * - multiply the (normalized) coordinates by the passed length
+   * - offset by the target coordinates
+   *
+   * @param {Array<Point>} points
+   * @param {ArrowData} arrowData
+   * @static
+   */
+  static transform(points, arrowData) {
```
Comment (yotamberk) [inline lib/network/modules/components/edges/util/EndPoints.js]: static won't work on browsers that don't support ES6

Comment (wimrijnders) [reply lib/network/modules/components/edges/util/EndPoints.js]: Well, we *do* use babel to convert `es2015 syntax` to 'regular' javascript. It's this line in the `package.json`:

```
    "babel-preset-es2015": "^6.24.1",
```

All `es2015` (aka `ES6`) syntax is transpiled to 'regular' javascript upon compilation with `gulp`. To give an example:

```js
class Node {
...
  static parseOptions(parentOptions, newOptions, allowDeletion = false, globalOptions = {}) {
...
  }
};
```

... becomes in the target `vis.js`:

```js
  (0, _createClass3['default'])(Node, [{
...
  }], [{
    key: 'parseOptions',
    value: function parseOptions(parentOptions, newOptions) {
      var allowDeletion = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : false;
      var globalOptions = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : {};
...
  }]);
  return Node;
}();
```

`static` just plain disappears in the target file `vis.js` We've been done the transpilation for some time now. I really think we're OK here. 

----

...if I'm missing something, please enlighten me.

Comment (yotamberk) [reply lib/network/modules/components/edges/util/EndPoints.js]: Ok, you're right again =)

**Pull request merged**